xbmc/platform/darwin/osx/XBMCApplication.mm

```diff
@@ -0,0 +1,390 @@
+  menuItem = [[NSMenuItem alloc] initWithTitle:@"Float on Top"
+                                        action:@selector(floatOnTopToggle:)
+                                 keyEquivalent:@"t"];
+  menuItem.keyEquivalentModifierMask = NSEventModifierFlagCommand;
+  [windowMenu addItem:menuItem];
+
+  // "Minimize" item
+  menuItem = [[NSMenuItem alloc] initWithTitle:@"Minimize"
+                                        action:@selector(performMiniaturize:)
+                                 keyEquivalent:@"m"];
+  menuItem.keyEquivalentModifierMask = NSEventModifierFlagCommand;
+  [windowMenu addItem:menuItem];
+
+  return windowMenu;
+}
+
+// The main class of the application, the application's delegate
+@implementation XBMCDelegate
+
+// Set the working directory to the .app's parent directory
+//! @todo Whats this for, is it required?
+- (void)setupWorkingDirectory
+{
+  auto parentPath = NSBundle.mainBundle.bundlePath.stringByDeletingLastPathComponent;
+  NSAssert([NSFileManager.defaultManager changeCurrentDirectoryPath:parentPath], @"SetupWorkingDirectory Failed to cwd");
+}
+
+- (void)stopRunLoop
+{
+  // to get applicationShouldTerminate and
+  // applicationWillTerminate notifications.
+  [NSApplication.sharedApplication terminate:nil];
+  // to flag a stop on next event.
+  [NSApplication.sharedApplication stop:nil];
+
+  //post a NOP event, so the run loop actually stops
+  //see http://www.cocoabuilder.com/archive/cocoa/219842-nsapp-stop.html
+  NSEvent* event = [NSEvent otherEventWithType:NSEventTypeApplicationDefined
+                                      location:NSMakePoint(0, 0)
+                                 modifierFlags:0
+                                     timestamp:0.0
+                                  windowNumber:0
+                                       context:nil
+                                       subtype:0
+                                         data1:0
+                                         data2:0];
+
+  [NSApplication.sharedApplication postEvent:event atStart:true];
+}
+
+// To use Cocoa on secondary POSIX threads, your application must first detach
+// at least one NSThread object, which can immediately exit. Some info says this
+// is not required anymore, who knows ?
+- (void)kickstartMultiThreaded:(id)arg
+{
+  @autoreleasepool
+  {
+    // empty
+  }
+}
+
+- (void)mainLoopThread:(id)arg
+{
+
+#if defined(DEBUG)
+  struct rlimit rlim;
+  rlim.rlim_cur = rlim.rlim_max = RLIM_INFINITY;
+  if (setrlimit(RLIMIT_CORE, &rlim) == -1)
+    CLog::Log(LOGDEBUG, "Failed to set core size limit (%s)", strerror(errno));
+#endif
+
+  setlocale(LC_NUMERIC, "C");
+
+  CAppParamParser appParamParser;
+  appParamParser.Parse((const char**)gArgv, (int)gArgc);
+
+  XBMC_Run(true, appParamParser);
+
+#ifdef _DEBUG
+  CServiceBroker::GetLogging().SetLogLevel(LOG_LEVEL_DEBUG);
+#else
+  //  CServiceBroker::GetLogging().SetLogLevel(LOG_LEVEL_NORMAL);
+  CServiceBroker::GetLogging().SetLogLevel(LOG_LEVEL_DEBUG);
```

Comment:
Is this intentional?

Reply:
Honestly, no idea. Carried over from existing master. It'll be another one of those, dont want to touch in this pr things. Can look at in a followup